drivers/input/input_analog_axis.c

```diff
@@ -0,0 +1,272 @@
+			    axis_cfg->out_min, axis_cfg->out_max);
+
+		if (cal->out_deadzone > 0) {
+			int16_t center = DIV_ROUND_CLOSEST(
+					axis_cfg->out_max  + axis_cfg->out_min, 2);
+			if (abs(out - center) < cal->out_deadzone) {
+				out = center;
+			}
+		}
+
+		if (axis_data->last_out != out) {
+			input_report_abs(dev, axis_cfg->axis, out, true, K_FOREVER);
+		}
+		axis_data->last_out = out;
+	}
+
+	k_mutex_unlock(&data->cal_lock);
+}
+
+static void analog_axis_thread(void *arg1, void *arg2, void *arg3)
+{
+	const struct device *dev = arg1;
+	const struct analog_axis_config *cfg = dev->config;
+	struct analog_axis_data *data = dev->data;
+	int err;
+	int i;
+
+	for (i = 0; i < cfg->num_channels; i++) {
+		const struct analog_axis_channel_config *axis_cfg = &cfg->channel_cfg[i];
+
+		if (!adc_is_ready_dt(&axis_cfg->adc)) {
+			LOG_ERR("ADC controller device not ready");
+			return;
+		}
+
+		err = adc_channel_setup_dt(&axis_cfg->adc);
+		if (err < 0) {
+			LOG_ERR("Could not setup channel #%d (%d)", i, err);
+			return;
+		}
+	}
+
+	k_timer_init(&data->timer, NULL, NULL);
+	k_timer_start(&data->timer,
+		      K_MSEC(cfg->poll_period_ms), K_MSEC(cfg->poll_period_ms));
+
+	while (true) {
+		analog_axis_loop(dev);
+		k_timer_status_sync(&data->timer);
+	}
+}
+
+static int analog_axis_init(const struct device *dev)
+{
+	struct analog_axis_data *data = dev->data;
+	k_tid_t tid;
+
+	k_mutex_init(&data->cal_lock);
+
+	tid = k_thread_create(&data->thread, data->thread_stack,
+			      K_KERNEL_STACK_SIZEOF(data->thread_stack),
+			      analog_axis_thread, (void *)dev, NULL, NULL,
+			      CONFIG_INPUT_ANALOG_AXIS_THREAD_PRIORITY,
+			      0, K_NO_WAIT);
+	if (!tid) {
+		LOG_ERR("thread creation failed");
+		return -ENODEV;
+	}
+
+	k_thread_name_set(&data->thread, dev->name);
+
+	return 0;
+}
+
+#define ANALOG_AXIS_CHANNEL_CFG_DEF(node_id) \
+	{ \
+		.adc = ADC_DT_SPEC_GET(node_id), \
+		.out_min = (int16_t)DT_PROP(node_id, out_min), \
+		.out_max = (int16_t)DT_PROP(node_id, out_max), \
+		.axis = DT_PROP(node_id, zephyr_axis), \
+		.invert = DT_PROP(node_id, invert), \
+	}
+
+#define ANALOG_AXIS_CHANNEL_CAL_DEF(node_id) \
+	{ \
+		.in_min = (int16_t)DT_PROP(node_id, in_min), \
+		.in_max = (int16_t)DT_PROP(node_id, in_max), \
+		.out_deadzone = DT_PROP(node_id, out_deadzone), \
+	}
+
+#define ANALOG_AXIS_INIT(inst) \
+	static const struct analog_axis_channel_config analog_axis_channel_cfg_##inst[] = { \
+		DT_INST_FOREACH_CHILD_STATUS_OKAY_SEP(inst, ANALOG_AXIS_CHANNEL_CFG_DEF, (,)) \
+	}; \
+	\
+	static struct analog_axis_channel_data \
+		analog_axis_channel_data_##inst[ARRAY_SIZE(analog_axis_channel_cfg_##inst)]; \
+	\
+	static struct analog_axis_calibration \
+		analog_axis_calibration##inst[ARRAY_SIZE(analog_axis_channel_cfg_##inst)] = { \
+		DT_INST_FOREACH_CHILD_STATUS_OKAY_SEP(inst, ANALOG_AXIS_CHANNEL_CAL_DEF, (,)) \
+		}; \
+	\
+	static const struct analog_axis_config analog_axis_cfg_##inst = { \
```

Comment:
maybe clang-format here?

Reply:
I'm afraid of the what the result could be, lately I'm pushing people off it for macro blocks more often than not, I'll indent the tabs though, I guess that's what the comment was about.